File: app/src/main/java/com/sjn/taggingplayer/constant/RepeatState.java
package com.sjn.taggingplayer.constant;

public enum RepeatState implements QueueState {
    ALL(0),
    ONE(1),
    NONE(2);
    final int mNo;

    RepeatState(int no) {
        mNo = no;
    }

    public int getNo() {
        return mNo;
    }

    public RepeatState toggle() {
        int no = (mNo + 1) % RepeatState.values().length;
        return RepeatState.of(no);
    }

    public static RepeatState of(int no) {
        for (RepeatState state : RepeatState.values()) {
            if (state.mNo == no) return state;
        }
        return null;
    }

    public static RepeatState getDefault() {
        return NONE;
    }
}

File: app/src/main/java/com/sjn/stamp/constant/RepeatState.kt
package com.sjn.stamp.constant

enum class RepeatState(val no: Int) : QueueState {
    ALL(0),
    ONE(1),
    NONE(2);

    fun toggle(): RepeatState {
        val no = (this.no + 1) % RepeatState.values().size
        return RepeatState.of(no)
    }

    companion object {

        fun of(no: Int): RepeatState = RepeatState.values().firstOrNull { it.no == no } ?: NONE

        fun getDefault(): RepeatState = RepeatState.NONE
    }
}
